src/main/java/org/altbeacon/beacon/service/GattBeaconTracker.java

```diff
@@ -69,13 +62,12 @@ private Beacon trackGattBeacon(Beacon beacon) {
     private void updateTrackingHashes(Beacon trackedBeacon, HashMap<Integer,Beacon> matchingTrackedBeacons) {
         if (matchingTrackedBeacons == null) {
             matchingTrackedBeacons = new HashMap<Integer,Beacon>();
-            mBeaconsByServiceUuidAndMac.put(serviceUuidAndMac(trackedBeacon), matchingTrackedBeacons);
         }
-        mBeacons.put(trackedBeacon.hashCode(), trackedBeacon);
         matchingTrackedBeacons.put(trackedBeacon.hashCode(), trackedBeacon);
+        mBeaconsByKey.put(getBeaconKey(trackedBeacon), matchingTrackedBeacons);
     }
 
-    private String serviceUuidAndMac(Beacon beacon) {
-        return beacon.getBluetoothAddress()+beacon.getServiceUuid();
+    private String getBeaconKey(Beacon beacon) {
+        return beacon.getBluetoothAddress() + beacon.getServiceUuid();
```

Comment:
If you remove the serviceUuid entirely, it will affect legacy behavior (e.g. it would associate Eddystone-TLM frames with an iBeacon or AltBeacon frame coming from the same hardware device.)  We probably don't want this.  So I don't think it is a good idea to remove it, but you could have it *conditionally* removed if the association were configured programmatically.